src/provider/model/EventPump.ts

```diff
@@ -0,0 +1,67 @@
+import {injectable, inject} from 'inversify';
+import {Identity} from 'openfin/_v2/main';
+
+import {Inject} from '../common/Injectables';
+import {APITopic, Events} from '../../client/internal';
+import {Transport, Targeted} from '../../client/EventRouter';
+import {NotificationActionEvent} from '../../client';
+
+import {APIHandler} from './APIHandler';
+import {ClientHandler} from './ClientHandler';
+
+type DeferrableEvent = Targeted<Transport<NotificationActionEvent>>;
+type DeferredEvent = {target: Identity, event: DeferrableEvent};
+
+/**
+ * Notification event message handler. EventPump internally handles the events added
+ * by either dispatching them immediately or deferring the dispatch.
+ *
+ */
+@injectable()
+export class EventPump {
+    @inject(Inject.API_HANDLER)
+    private _apiHandler!: APIHandler<APITopic, Events>;
```

Comment:
Can you inject this via the constructor too? We tend to use ctor injection by default.

Reply:
done